src/components/WebLogView/ContentPreview.tsx

```diff
@@ -0,0 +1,99 @@
+import { Flex, SegmentedControl, Box, ScrollArea } from '@radix-ui/themes'
+import { useState, useEffect } from 'react'
+
+import { ReadOnlyEditor } from '../Monaco/ReadOnlyEditor'
+
+import { useGoToContentMatch } from './Details.hooks'
+import { Preview } from './ResponseDetails/Preview'
+import { Raw } from './ResponseDetails/Raw'
+
+type ContentPreviewProps = {
+  content: string
+  contentType: string
+  format: string
+  rawContent: string | undefined
+}
+
+export function ContentPreview({
+  format,
```

Comment:
This component was extracted from the existing `Content` component to be reused in `Payload` as well.